tests/system_tests_skmanage.py

```diff
@@ -179,15 +179,24 @@ def long_type(name):
         self.assertEqual([long_type('listener')] * 4, [e['type'] for e in qlistener])
         self.assertEqual(self.router_1.ports[0], int(qlistener[0]['port']))
 
-        qattr = json.loads(self.run_skmanage('query type name'))
-        for e in qattr:
-            self.assertEqual(2, len(e))
-
         def name_type(entities):
             ignore_types = [long_type(t) for t in ['router.link', 'connection', 'router.address']]
             return set((e['name'], e['type']) for e in entities
                        if e['type'] not in ignore_types)
-        self.assertEqual(name_type(qall), name_type(qattr))
+
+        test_passed = False
+        for i in range(10):
+            qattr = json.loads(self.run_skmanage('query type name'))
+            try:
+                for e in qattr:
+                    self.assertEqual(2, len(e))
+                self.assertEqual(name_type(qall), name_type(qattr))
+                test_passed = True
+                break
+            except AssertionError as ae:
+                # Give some time for the Router 1's data structures to update.
+                sleep(0.5)
```

Comment:
Just passing through! But this sort of code is just asking for a flaky test, In a year or under some heavy load or on a different system 5s (10*0.5s) will not be long enough and the test will fail.
Please, please, pretty please architect systems so that you can test them without arbitrary sleeps!
I expect this is not going to be an issue here, but I'm triggered! /rant

Reply:
There will always be some arbitrariness in this. The current arbitrary (total) timeout of choice for waiting (the `TIMEOUT` constant) is 60 seconds. That's why I like `system_tests#retry_exception`, because it defaults to using this timeout.

Reply:
The repeated querying is necessary in this case to find out if the router has the expected state. The setupClass function makes sure that both routers are connected to each other by executing a specific management query on both routers but this not guarantee that everything is up-to-date in each router. We need a good way for the router(s) to tell the world (if asked) that the entire router network is completely ready to go so the clients would not have to retry to find out if the routers have reached steady state.

Reply:
Nobody's disputing the need for a retry with a timeout, IIUC. The only objection is that 5 second timeout might be too short.

I think you should wait up to `TIMEOUT` seconds in total, as sum of all waits of all iterations.